src/main/scala/is/hail/io/plink/PlinkLoader.scala

```diff
@@ -200,8 +137,59 @@ object PlinkLoader {
            |  Duplicate IDs: @1""".stripMargin, duplicateIds)
     }
 
-    val vds = parseBed(hc, bedPath, ids, annotations, signature, variants, nPartitions, a2Reference)
-    vds
-  }
+    val results = bedPaths.zip(bimPaths).map { case (bedPath, bimPath) =>
```

Comment:
This executes serially, which will take a huge amount of time for large numbers of files. At 1s per file (which seems like a lower bound on the cloud to read + broadcast the bim), then this will take an hour for 3600 partitions, which may not be so uncommon.

I'm not entirely sure what to do, though. Our current strategy makes it hard to parallelize this chunk.

Reply:
Let's think on this and discuss in person tomorrow.